File: 1616-minimum-difference-between-largest-and-smallest-value-in-three-moves/minimum-difference-between-largest-and-smallest-value-in-three-moves.py
class Solution:
    def minDifference(self, nums: List[int]) -> int:
        nums.sort()
        if len(nums)<=4:
            return 0
        else:
            ans=nums[len(nums)-1]-nums[0]
            for i in range(0,4):
                ans=min(ans,nums[len(nums)-1-i]-nums[3-i])
            return ans

File: 1616-minimum-difference-between-largest-and-smallest-value-in-three-moves/minimum-difference-between-largest-and-smallest-value-in-three-moves.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minDifference = function(nums) {
    nums.sort((a,b)=>a-b);
    if(nums.length<=4){
        return 0;
    }
    else{
        var ans=nums[nums.length-1]-nums[0];
        // console.log(ans);
        for(var i=0;i<4;i++){
            ans=Math.min(ans,nums[nums.length-1-i]-nums[3-i]);
        }
        return ans;
    }

};